rubin_scheduler/utils/consdb.py

```diff
@@ -316,8 +316,32 @@ def consdb_visits(self) -> pd.DataFrame:
         prior_day_obs_date = day_obs_date - timedelta(days=self.num_nights)
         prior_day_obs_int = int(prior_day_obs_date.strftime("%Y%m%d"))
 
+        # To avoid duplicate columns, explicitly include list of columns to
+        # to return in the query instead of using a "SELECT *".
+        def columns_query(table, instrument):
+            return f"""
+                SELECT column_name FROM information_schema.columns
+                WHERE table_name = '{table}'
+                  AND table_schema = 'cdb_{instrument}'
+            """
+
+        exposure_columns = (
+            query_consdb(columns_query("exposure", self.instrument)).loc[:, "column_name"].values
+        )
+        ql_columns = (
+            query_consdb(columns_query(f"visit{self.num_exposures}_quicklook", self.instrument))
+            .loc[:, "column_name"]
+            .values
+        )
+        visit_query_exposure_columns = ", ".join([f"e.{c} AS {c}" for c in exposure_columns])
+        visit_query_ql_columns = ", ".join(
+            [f"v{self.num_exposures}q.{c} AS {c}" for c in ql_columns if c not in exposure_columns]
+        )
+        visit_query_columns = ", ".join([visit_query_exposure_columns, visit_query_ql_columns])
+
         consdb_visits_query: str = f"""
-            SELECT * FROM cdb_{self.instrument}.exposure AS e
+            SELECT {visit_query_columns}
```

Comment:
These aren't visit columns if you're querying the exposure table, they're exposure columns.

Reply:
That's because some of the columns we need is present in the exposure table and not the visit table. I was when starting this for OR4 that joining the visit_id to the exposure_id would get the first exposure, which was good enough for what I needed (particularly given that for OR4 there was only one exposure per visit).

For arbitrary numbers of exposures per visit, the best thing is probably to find some kind of a join that would match all exposure corresponding to a visit to that visit, GROUP BY visit_id, and use SQL aggregation functions (probably just using the first exposure in each group). For OR4, such a join wasn't possible.

Double checking, it looks ilke it still isn't possible to do this join correctly, so multiple exposures match each visit. But, there still isn't a visit2 table, so I don't think multi-snap visits are supported by the consdb yet anyway.

However, it looks like most of the columns of interest are present in the visits1 table now anyway, so maybe the join to exposure is no longer necessary. I'll give that a try.

Reply:
If there are things which don't work about the consdb as it is now, I think the answer is probably to try to get it changed -- or at the very least, record and pass along what isn't working.

Reply:
Whatever the issues was with the visits table in OR4 that drove me to use exposure instead, it seems to be solved now: I switched from using `exposures` to` f"visits{self.num_exposures}"`, and it seems to be working.